CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKCommandBuffer.java

```diff
@@ -236,103 +215,100 @@ private void beginRenderPassCmd(long renderPass, long frameBuffer,
-		pVertexBuffers.put(0, vertexBuffer);
-		
-		vkCmdBindVertexBuffers(vkCommandBuffer, 0, pVertexBuffers, offsets);
-		
-		memFree(pVertexBuffers);
-		memFree(offsets);
-	}
+    public void PushConstants(long pipelineLayout, int stageFlags, int offset, ByteBuffer data) {
+
+        vkCmdPushConstants(vkCommandBuffer, // The buffer to push the matrix to
+                        pipelineLayout,     // The pipeline layout
+                        stageFlags,         // Flags
+                        offset,             // Offset
+                        data);              // Push constant data
+    }
 	
-	public void bindComputeDescriptorSetsCmd(long pipelinyLayout, long[] descriptorSets){
+    public void BindVertexInput(long vertexBuffer, long indexBuffer) {
+        
+        try (MemoryStack stack = stackPush()) {
+            LongBuffer pVertexBuffers = stack.longs(vertexBuffer);
+            LongBuffer offsets = stack.longs(0);
 
-		bindDescriptorSetsCmd(pipelinyLayout, descriptorSets,
-				VK_PIPELINE_BIND_POINT_COMPUTE);
-	}
-//	
-//	public void bindGraphicsDescriptorSetsCmd(long pipelinyLayout, long[] descriptorSets){
-//		
-//		bindDescriptorSetsCmd(pipelinyLayout, descriptorSets,
-//				VK_PIPELINE_BIND_POINT_GRAPHICS);
-//	}
-//	
-	private void bindDescriptorSetsCmd(long pipelinyLayout, long[] descriptorSets,
-			int pipelineBindPoint){
-		
-		vkCmdBindDescriptorSets(vkCommandBuffer, pipelineBindPoint,
-				pipelinyLayout, 0, descriptorSets, null);
-	}
+            vkCmdBindVertexBuffers(vkCommandBuffer, 0, pVertexBuffers, offsets);
+            vkCmdBindIndexBuffer(vkCommandBuffer, indexBuffer, 0, VK_INDEX_TYPE_UINT32);
+        }
+    }
+	
+    public void BindVertexInput(long vertexBuffer) {
+        
+        try (MemoryStack stack = stackPush()) {
+            
+            LongBuffer pVertexBuffers = stack.longs(vertexBuffer);
+            LongBuffer offsets = stack.longs(0);
+            
+            // Bind verts
+            vkCmdBindVertexBuffers(vkCommandBuffer, 0, pVertexBuffers, offsets);
+            
+        }
+    }
+	
+    public void BindComputeDescriptorSets(long pipelinyLayout, long descriptorSets) {
+
+        BindDescriptorSets(pipelinyLayout, descriptorSets,
+                            VK_PIPELINE_BIND_POINT_COMPUTE);
+    }
+	
+    public void BindGraphicsDescriptorSets(long pipelinyLayout, long descriptorSets){
+
+        BindDescriptorSets(pipelinyLayout, descriptorSets,
+                            VK_PIPELINE_BIND_POINT_GRAPHICS);
+    }
+
+    private void BindDescriptorSets(long pipelinyLayout, long descriptorSets, int pipelineBindPoint){
+        
+        try (MemoryStack stack = stackPush()) {
+            vkCmdBindDescriptorSets(vkCommandBuffer, pipelineBindPoint,
+                            pipelinyLayout, 0, stack.longs(descriptorSets), null);
+        }
+    }
 	
-//	public void clearColorImageCmd(long image, int imageLayout){
+//	public void ClearColorImage(long image, int imageLayout){
 //		
```

Comment:
Remove?  Or at least spell it colour....

Reply:
I thought the convention in Constellation was to use Color